src/hdx_cli/cli_interface/role/commands.py

```diff
@@ -0,0 +1,325 @@
+import json
+import uuid
+import click
+
+from ...library_api.common.exceptions import (LogicException,
+                                              InvalidRoleException,
+                                              ResourceNotFoundException)
+from ...library_api.common.role_validator import (Role, Policy,
+                                                  get_role_data_from_standard_input,
+                                                  modify_role_data_from_standard_input,
+                                                  update_role_request)
+from ...library_api.userdata.token import AuthInfo
+from ...library_api.common import rest_operations as rest_ops
+from ...library_api.utility.decorators import report_error_and_exit
+from ...library_api.common.context import ProfileUserContext
+from ..common.undecorated_click_commands import basic_create_with_body_from_string, basic_show
+from ..common.rest_operations import (delete as command_delete,
+                                      list_ as command_list,
+                                      show as command_show)
+
+
+@click.group(help="Role-related operations")
+@click.option('--role', 'role_name', help='Perform operation on the passed role.',
+              metavar='ROLE_NAME', default=None)
+@click.pass_context
+def role(ctx: click.Context,
+         role_name):
+    user_profile = ctx.parent.obj['usercontext']
+    ctx.obj = {'resource_path': '/config/v1/roles/',
+               'usercontext': user_profile}
+    ProfileUserContext.update_context(user_profile,
+                                      rolename=role_name)
+
+
+AVAILABLE_SCOPE_TYPE = ('user', 'pool', 'role', 'invite', 'org', 'project',
+                        'alterjob', 'batchjob', 'catalog', 'hdxstorage',
+                        'dictionary', 'dictionaryfile', 'function', 'table',
+                        'kafkasource', 'siemsource', 'kinesissource',
+                        'summarysource', 'transform', 'view')
```

Comment:
There is an API endpoint `/roles/permissions/` that will list all the possible scope types and all the possible permissions that go with each scope type. Hard-coding like this is fine since it's not likely to change soon. But it would be better to dynamically load the scope types from the API. That will make sure that if a user has an older hdxcli version it will be more compatible with different Hydrolix versions.